pkg/resiliency/breaker/circuitbreaker.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package breaker
+
+import (
+	"time"
+
+	"github.com/sony/gobreaker"
+
+	"github.com/dapr/dapr/pkg/expr"
+)
+
+type CircuitBreaker struct {
+	Name        string
+	MaxRequests uint32        `mapstructure:"maxRequests"`
+	Interval    time.Duration `mapstructure:"interval"`
+	Timeout     time.Duration `mapstructure:"timeout"`
+	Trip        *expr.Expr    `mapstructure:"trip"`
+	breaker     *gobreaker.CircuitBreaker
+}
+
+func IsErrorPermanent(err error) bool {
+	return err == gobreaker.ErrOpenState || err == gobreaker.ErrTooManyRequests
```

Comment:
How can components emit permanent errors too?

Reply:
Wrap the error with `backoff.Permanent(err)`

Reply:
This can be wrapped in `kit/retry` to hide the implementations but ultimately it needs to be `backoff.Permanent` and not an interface of some sort.

https://github.com/cenkalti/backoff/blob/6b0e4ad0cd65431b217393bff47d1dff727b264b/retry.go#L59-L62

Reply:
This will need to be incorporated for the pluggable component later on since the response will come from another process.